### DISCUSSION TITLE
Need an example of an Oracle Plugin Adapter

### DISCUSSION THREAD

**Discussion by danielkboyer** [metaplex-foundation/mpl-core#143] [General]
From Metaplex documentation:

> Example 2
> Assets can only be updated if the floor price is above $10 and the asset has attribute “red hat”.
> 
> Create onchain Oracle Plugin in a program of your choice.
> Add the Oracle Plugin Adapter to Asset specifying the lifecycle events you wish to have rejection validation over.
> Dev writes Anchor program that can write to the Oracle Account that derive the same PRECONFIGURED_ASSET accounts
> Dev writes web2 script that watches prices on a marketplace, AND with known hashlist of Assets with the 'Red Hat' trait red updates and writes to the relevant Oracle Accounts.

It would be very helpful if there was an example of an Oracle Plugin Adapter.

**Comment by WangWilly** [comment]
Hi @danielkboyer , here is my project which is applied oracle plugin. you can see how I use it. However, I can't guarantee my usage is the bast practice. https://github.com/WangWilly/solana-poneglyph/tree/master/contract
